### PR TITLE
upload results as artifact and/or security tab

### PR DESCRIPTION
**Pull request by ale8k**
We need to run security scans on a specified branch ideally weekly. This emulates CodeQL provided by GitHub and is inline with what other teams in canonical are doing. This workflow allows us to use trivy across repos and for those with QL enabled, upload there, if not, send them to artifacts on the run.

This will be used in a CRON workflow in each projects repo to check for vulnerabilities that arise in the main / released branch(s).

### CODE REVIEW

File: .github/actions/security-scan/action.yaml [hunk 1]
```diff
@@ -1,6 +1,17 @@
 name: Security Scan
 description: "Run Go vulernability scan and Trivy repo scan"
 
+inputs:
+  upload_results:
+    description: 'Whether to upload scan results to GitHub Security tab'
```
Comment (babakks) [inline .github/actions/security-scan/action.yaml:6]: You don't need quotes here.

**Review by babakks — APPROVED** [review]

File: .github/actions/security-scan/action.yaml [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+    required: false
+    default: 'false'
```
[the remark this answers is not in the document]
Comment (ale8k) [reply .github/actions/security-scan/action.yaml:8]: I couldn't see a way to use anything else as an input to a composite action?

File: .github/actions/security-scan/action.yaml [hunk 3]
```diff
@@ -16,5 +27,20 @@ runs:
         format: 'table' 
         severity: 'CRITICAL'
         exit-code: '1'
+        output: 'trivy-results.sarif'
```
Comment (kian99) [inline .github/actions/security-scan/action.yaml:30]: We can do the same thing with the go vuln check results. (a little above this)

**Review by kian99 — COMMENTED** [review]
Looks good but we can do the same thing for the Go vulnerability scan, it supports sarif output format.

File: .github/actions/security-scan/action.yaml (hunk 3 above)
Comment (ale8k) [reply .github/actions/security-scan/action.yaml:30]: I was thinking this but I was wondering what uploading two will result in on GH Security Tab... I guess as an artifact nothing?

Comment (kian99) [reply .github/actions/security-scan/action.yaml:30]: No clue tbh

**Review by kian99 — APPROVED** [review]
approving regardless of the go scan bits

**Pull request merged**